modules/openapi-generator-maven-plugin/src/main/java/org/openapitools/codegen/plugin/CodeGenMojo.java

```diff
@@ -801,9 +801,7 @@ private String calculateInputSpecHash(File inputSpecFile) throws IOException {
     private URL inputSpecRemoteUrl(){
         try {
             return new URI(inputSpec).toURL();
-        } catch (URISyntaxException e) {
-            return null;
-        } catch (MalformedURLException e) {
+        } catch (URISyntaxException | MalformedURLException | IllegalArgumentException e) {
```

Comment:
This method is to get the URL if remote otherwise null.
when inputSpec is local file and relative path, IllegalArgumentException is thrown.
In this case, we can return null because this method is expected to return URL only when the inputSpec is remote.